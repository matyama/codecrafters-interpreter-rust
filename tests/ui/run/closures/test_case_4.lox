fun makeAccumulator(label) {
  var sum = 0;
  var count = 0;

  fun accumulate(value) {
    sum = sum + value;
    count = count + 1;

    print label;
    print count;
    print sum;
    print sum;

    if (count > 3) {
      print "reset";
      sum = 0;
      count = 0;
    }

    return sum;
  }

  return accumulate;
}

var acc1 = makeAccumulator("First:");
var acc2 = makeAccumulator("Second:");

acc1(2);
acc1(6);
acc1(3);
acc1(4);

acc2(6);
acc2(4);
