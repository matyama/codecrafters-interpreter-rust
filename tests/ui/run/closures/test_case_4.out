First:
1
2
2
First:
2
8
8
First:
3
11
11
First:
4
15
15
reset
Second:
1
6
6
Second:
2
10
10
